### ISSUE TITLE
Add accessor functions for ochttp's addedTagsKey

### ISSUE THREAD

**Issue by basvanbeek** [census-instrumentation/opencensus-go#944]
**Is your feature request related to a problem? Please describe.**

The current route tagging logic only works for routers supporting `http.Handler`. Some routers, like the one from gin do not support this. Because of this a large part of ochttp would need to be copied to a specific gin middleware (as demonstrated by @sagikazarmark in  https://github.com/sagikazarmark/go-gin-gorm-opencensus). 

**Describe the solution you'd like**
To allow usage of standard ochttp on the outer shell with tag.Mutator access from within gin handlers we could provide standard context accessors for the addedTags slice and make gin integration much easier with our standard plugin.

**Comment by sagikazarmark** [comment]
That wouldn't solve the problem as the route would only become available later:

ochttp -> custom routing strategy -> add route info to span

**Comment by ramonza** [comment]
What if we extracted the logic for HTTP request processing, something like:

```go

func (h *ochttp.Handler)  Start(req *http.Request) *http.Request {
...
  return req.WithContext(newCtx)
}

func (h *ochttp.Handler) End(req *http.Request, route string) {
  span := trace.FromContext(req)
  span.AddAttribute(trace.StringAttribute("http.route", route))
  // add route as tag and call stats.RecordWithTag
}

```

Then frameworks could construct an appropriate http.Request and call Start and End directly (rather than relying on Handle).

**Comment by sagikazarmark** [comment]
That could work, but it would also mean that ochttp cannot the be most outer layer of the http request, so that the amount of instrumented code would possibly be less.

What if we don't expose the whole tag adding layer, just add a method that adds the route to the tag list in the context?

**Comment by ramonza** [comment]
Yeah, that's fine I think. We should just note that it isn't safe to call it from a separate goroutine. Can we call it `ochttp.SetRoute(req, route)` and have it set the `http.route` attribute on the associated span as well?

**Comment by sagikazarmark** [comment]
I think it should accept and return a context. Context propagation is not necessarily tied to an http request within the framework boundaries.

What should it do if a tag list is not available in the context? Create it?

**Comment by sagikazarmark** [comment]
> Come to think of it, the function should probably _not_ return a context, since it isn't adding anything to the context but rather internally mutating it.

I agree, it was only an option if we decided to add tag list to the context if it doesn't exist.

The signature looks good.
